package/lib/compileFunctions.test.js

```diff
@@ -767,4 +767,582 @@ describe('CompileFunctions', () => {
+      expect(
+        googlePackage.serverless.service.provider.compiledConfigurationTemplate.resources
+      ).toEqual(compiledResources);
+    });
+  });
+
+  it('should allow vpc egress all at provider level', () => {
+    googlePackage.serverless.service.functions = {
+      func1: {
+        handler: 'func1',
+        memorySize: 128,
+        runtime: 'nodejs10',
+        vpc: 'my-vpc',
+        events: [{ http: 'foo' }],
+      },
+    };
+
+    googlePackage.serverless.service.provider.vpcEgress = 'all';
+
+    const compiledResources = [
+      {
+        type: 'gcp-types/cloudfunctions-v1:projects.locations.functions',
+        name: 'my-service-dev-func1',
+        properties: {
+          parent: 'projects/myProject/locations/us-central1',
+          runtime: 'nodejs10',
+          function: 'my-service-dev-func1',
+          entryPoint: 'func1',
+          availableMemoryMb: 128,
+          timeout: '60s',
+          sourceArchiveUrl: 'gs://sls-my-service-dev-12345678/some-path/artifact.zip',
+          httpsTrigger: {
+            url: 'foo',
+          },
+          labels: {},
+          vpcConnector: 'my-vpc',
+          vpcConnectorEgressSettings: 'ALL_TRAFFIC',
+        },
+      },
+    ];
+
+    return googlePackage.compileFunctions().then(() => {
+      expect(consoleLogStub.called).toEqual(true);
+      expect(
+        googlePackage.serverless.service.provider.compiledConfigurationTemplate.resources
+      ).toEqual(compiledResources);
+    });
+  });
+
+  it('should allow vpc egress private at provider level', () => {
+    googlePackage.serverless.service.functions = {
+      func1: {
+        handler: 'func1',
+        memorySize: 128,
+        runtime: 'nodejs10',
+        vpc: 'my-vpc',
+        events: [{ http: 'foo' }],
+      },
+    };
+
+    googlePackage.serverless.service.provider.vpcEgress = 'private';
+
+    const compiledResources = [
+      {
+        type: 'gcp-types/cloudfunctions-v1:projects.locations.functions',
+        name: 'my-service-dev-func1',
+        properties: {
+          parent: 'projects/myProject/locations/us-central1',
+          runtime: 'nodejs10',
+          function: 'my-service-dev-func1',
+          entryPoint: 'func1',
+          availableMemoryMb: 128,
+          timeout: '60s',
+          sourceArchiveUrl: 'gs://sls-my-service-dev-12345678/some-path/artifact.zip',
+          httpsTrigger: {
+            url: 'foo',
+          },
+          labels: {},
+          vpcConnector: 'my-vpc',
+          vpcConnectorEgressSettings: 'PRIVATE_RANGES_ONLY',
+        },
+      },
+    ];
+
+    return googlePackage.compileFunctions().then(() => {
+      expect(consoleLogStub.called).toEqual(true);
+      expect(
+        googlePackage.serverless.service.provider.compiledConfigurationTemplate.resources
+      ).toEqual(compiledResources);
+    });
+  });
+
+  it('should replace vpc egress private at provider level for a vpc egress all defined at function level', () => {
```

Comment:
I think we need just a single test for `provider` being overrided by `function` level setting. These further tests are not really needed so let's remove them